src/libostree/ostree-repo-prune.c

```diff
@@ -515,10 +567,23 @@ ostree_repo_prune_from_reachable (OstreeRepo        *self,
     return FALSE;
 
   g_autoptr(GHashTable) objects = NULL;
-
-  if (!ostree_repo_list_objects (self, OSTREE_REPO_LIST_OBJECTS_ALL | OSTREE_REPO_LIST_OBJECTS_NO_PARENTS,
+  OstreeRepoPruneFlags flags = options->flags;
+  gboolean list_refs_only = flags & OSTREE_REPO_PRUNE_FLAGS_LIST_REFS_ONLY;
```

Comment:
Very minor nit but I usually coerce bitflags to true booleans via e.g. `(flags & OSTREE_REPO_PRUNE_FLAGS_LIST_REFS_ONLY) > 0` or so; today `gboolean` is just a typedef for `int` and in C, anything non-zero is true...but it can be confusing to have its value be `4` or whatever.  Modern languages like Rust don't have this implicit coercion and have a true `bool` type and e.g. the [bitflags crate has contains()](https://docs.rs/bitflags/1.3.2/bitflags/example_generated/struct.Flags.html#method.contains) which is IMO definitely better.